dvc/repo/add.py

```diff
@@ -65,6 +65,12 @@ def add(repo, targets, recursive=False, no_commit=False, fname=None):
 
 
 def _find_all_targets(repo, target, recursive):
+    target = os.path.normpath(target)
+    if os.path.commonprefix(
+        [os.path.abspath(target), repo.root_dir]
+    ).startswith(repo.root_dir):
```

Comment:
Use `path_isin()`. Also this code allows to include all repo recursively, not sure this is desirable.